src/main/groovy/de/iteratec/osm/result/dao/EventResultQueryBuilder.groovy

```diff
@@ -370,26 +373,39 @@ class EventResultQueryBuilder {
         userTimingQueryExecutor.setTransformer(new UserTimingAverageDataTransformer(baseProjections: baseProjections))
         userTimingQueryExecutor.setTrimmer(new UserTimingDataTrimmer())
 
-        return getResults()
+        return getResults(false)
     }
 
-    private List<EventResultProjection> getResults() {
+    private List<EventResultProjection> getResults(boolean minimumLevel) {
+        List<Map> userTimingRawData = []
+        List<Map> measurandRawData = []
+        List<EventResultProjection> userTimingsResult = []
+        List<EventResultProjection> measurandResult = []
 
-        List<EventResultProjection> userTimingsResult = this.performanceLoggingService.logExecutionTimeSilently(
-                PerformanceLoggingService.LogLevel.INFO, 'getResults - getting user timing results', 3) {
-            userTimingQueryExecutor.getResultFor(filters, trims, baseProjections, performanceLoggingService)
+        this.performanceLoggingService.logExecutionTimeSilently(
+                PerformanceLoggingService.LogLevel.INFO,
+                'getResults - getting user timing results',
+                3) {
+            userTimingRawData = userTimingQueryExecutor.getRawResultDataFor(filters, trims, baseProjections, performanceLoggingService)
+            userTimingsResult = userTimingQueryExecutor.getResultFor(userTimingRawData, performanceLoggingService)
         }
-        List<EventResultProjection> measurandResult = this.performanceLoggingService.logExecutionTimeSilently(
-                PerformanceLoggingService.LogLevel.INFO, 'getResults - getting measurand results', 3) {
-            measurandQueryExecutor.getResultFor(filters, trims, baseProjections, performanceLoggingService)
+        this.performanceLoggingService.logExecutionTimeSilently(
+                PerformanceLoggingService.LogLevel.INFO,
+                'getResults - getting measurand results',
+                3) {
+            measurandRawData = measurandQueryExecutor.getRawResultDataFor(filters, trims, baseProjections, performanceLoggingService)
+            measurandResult = measurandQueryExecutor.getResultFor(measurandRawData, performanceLoggingService)
         }
 
-        List<EventResultProjection> merged
-        performanceLoggingService.logExecutionTimeSilently(PerformanceLoggingService.LogLevel.INFO, "getResults - merge ${measurandResult.size()} measurand results with ${userTimingsResult.size()} user timing results", 3) {
-            merged = mergeResults(measurandResult, userTimingsResult)
-        }
+        if (!minimumLevel || userTimingRawData.size() + measurandRawData.size() >= 30) {
+            List<EventResultProjection> merged = []
+            performanceLoggingService.logExecutionTimeSilently(PerformanceLoggingService.LogLevel.INFO, "getResults - merge ${measurandResult.size()} measurand results with ${userTimingsResult.size()} user timing results", 3) {
```

Comment:
Please move the number `3` to a (class) constant for better readability.
Please check for more occurrences.